ezgui/src/event_ctx.rs

```diff
@@ -35,8 +52,17 @@ impl<'a> EventCtx<'a> {
         f(self, &mut timer)
     }
 
+    pub fn request_update(&mut self, update_type: UpdateType){
+        self.updates_requested.push(update_type);
+    }
+
     pub fn canvas_movement(&mut self) {
-        self.canvas.handle_event(&mut self.input)
+        match self.canvas.handle_event(&mut self.input) {
```

Comment:
`Option` is iterable like `Vec`, so you could just say: `self.updates_requested.extend(`self.canvas.handle_event(&mut self.input))`